harness/determined/cli/token.py

```diff
@@ -0,0 +1,182 @@
+        resp = bindings.get_GetAccessTokens(sess, username=username, showInactive=show_inactive)
+        if args.json or args.yaml:
+            json_data = [t.to_json() for t in resp.tokenInfo]
+            if args.json:
+                render.print_json(json_data)
+            else:
+                print(util.yaml_safe_dump(json_data, default_flow_style=False))
+        else:
+            render_token_info(resp.tokenInfo)
+    except Exception as e:
+        raise errors.CliError(f"Error fetching tokens: {e}")
+
+
+def revoke_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        request = bindings.v1PatchAccessTokenRequest(
+            tokenId=args.token_id, description=None, setRevoked=True
+        )
+        resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully revoked token {args.token_id}.")
+
+
+def create_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    username = args.username or sess.username
+    user = bindings.get_GetUserByUsername(session=sess, username=username).user
+
+    if user is None or user.id is None:
+        raise errors.CliError(f"User '{username}' not found or does not have an ID")
+
+    request = bindings.v1PostAccessTokenRequest(
+        userId=user.id, lifespan=args.expiration_duration, description=args.description
+    )
+    resp = bindings.post_PostAccessToken(sess, body=request).to_json()
+
+    output_string = None
+    if args.yaml:
+        output_string = util.yaml_safe_dump(resp, default_flow_style=False)
+    elif args.json:
+        output_string = json.dumps(resp, indent=2)
+    else:
+        output_string = f'TokenID: {resp["tokenId"]}\nAccess-Token: {resp["token"]}'
+
+    print(output_string)
+
+
+def edit_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        if args.description and args.token_id:
+            request = bindings.v1PatchAccessTokenRequest(
+                tokenId=args.token_id, description=args.description, setRevoked=False
+            )
+            resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.APIException as e:
+        raise errors.CliError(f"Caught APIException: {str(e)}")
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print(f"Successfully updated token with ID: {args.token_id}.")
+
+
+def login_with_token(args: argparse.Namespace) -> None:
+    unauth_session = api.UnauthSession(master=args.master, cert=cli.cert)
+    auth_headers = {"Authorization": f"Bearer {args.token}"}
+    user_data = unauth_session.get("/api/v1/me", headers=auth_headers).json()
```

Comment:
might be worth catching the exception here if token isn't valid.

Reply:
Will encapsulate them with try/except{UnauthenticatedException} block